src/supernova/snark.rs

```diff
@@ -207,6 +215,8 @@ where
     + (7 + 2 * pp.augmented_circuit_params_primary.get_n_limbs()); // # 1 * (7 + [X0, X1]*#num_limb)
 
     // secondary circuit
+    // NOTE: This count ensure the number of witnesses sent by the prover must bd equal to the number
+    // of NIVC circuits
     let num_field_secondary_ro = 2 // params_next, i_new
     + 2 * pp.circuit_shape_secondary.F_arity // zo, z1
     + pp.circuit_shapes.len() * (7 + 2 * pp.augmented_circuit_params_primary.get_n_limbs()); // #num_augment
```

Comment:
> This will error if a bad result has been stored in `res_primary`, but it won't catch the case in which some checks were never performed. Test case to consider: is it possible to provide input to this verification function such that `res_primary` is empty here? If so, would that input represent a valid proof? If not, would this verification function successfully 'fail' (by returning an error)?
> 
> It would be a good exercise and provide valuable confidence in verifier soundness to write one or more 'negative' tests, demonstrating that such maliciously-constructed `CompressedSNARK`s will not wrongly verify.
https://github.com/lurk-lab/arecibo/pull/80#discussion_r1373790940

(not intentionally, but...) This count for the RO initialization actually detects a proof that is invalid because the verifier is not given the right number of running instances. 

I have added a corresponding test to ensure that this is indeed happening.

Reply:
Ah, great. Thank you for clarifying that detail.

Reply:
My points above stand — in terms of UX and subtle probably-accidental implementation pitfalls; but I acknowledge that this check may make my point about verifier correctness inapplicable. Thanks for adding a test to demonstrate this.